src/linear_algebra/determinant.lean

```diff
@@ -452,12 +452,14 @@ multilinear map. -/
 def basis.det : alternating_map R M R ι :=
```

Comment:
Would it make sense to make this definition not depend on `decidable_eq iota`, by using the classical instance in `to_fun`? Normally, uses of this function should not unfold the `to_fun` or `to_matrix`, these are just implementation details.

Reply:
In the long term I'd like to make this computable too; the only reason it's not it due to silly `finsupp` reasons. At that point, the `decidable_eq ι` would need to stick around.

Of course, we're not at that point yet so `decidable_eq ι` is offering nothing for computability anyway and to that end could be removed. On the other hand, such a change is somewhat tangential to the main change in this PR, and just creates even more work for an eventual "make `basis` computational" refactor.

Since keeping the argument around is the status quo, I think I'd err on the side of not removing it in _this_ PR. If you feel strongly, I would begrudgingly approve a follow-up PR that removed it.

Reply:
If you have plans to  make this computable, it makes more sense to keep the `decidable_eq` around. So ok for the status quo.